source/Engine.cpp

```diff
@@ -2738,3 +2738,16 @@ void Engine::EmplaceStatusOverlay(const shared_ptr<Ship> &it, Preferences::Overl
 	statuses.emplace_back(it->Position() - center, it->Shields(), it->Hull(),
 		min(it->Hull(), it->DisabledHull()), max(20., width * .5), type, alpha);
 }
+
+
+
+void Engine::HandleEvents()
+{
+	for(const auto &event : events)
+		if(event.Type() & ShipEvent::DISABLE && event.Target()->CanBeCarried())
+			for(auto &projectile : projectiles)
+				if(projectile.Target() == event.Target().get())
+					projectile.BreakTarget();
```

Comment:
We should probably change this so that it only iterates over each collection once instead of iterating over the projectiles list every time.
1. Iterate through all events to find the ships we need to break targeting on. (Throw `event.Target()` into a `set<>`.)
2. Iterate through all the projectiles and call BreakTarget on those that have a target in the above set. (`set.contains(projectile.Target())` or something like that.)

Realistically you'd only get multiple disable events in the same frame with bigger battles, but it'd be cleaner to set it up this way should we ever have to handle more events.

Reply:
Even in bigger battles, a ship should only get disabled once in a frame, so we don't need it to be a `set<>`, but yeah, that would be cleaner.

Reply:
True. A vector would work just fine.

Reply:
I just realized that a set is still better because we are doing lookups...

Reply:
And I realized it's `count`, not `contains`.  That's what I get for regularly swapping between three programming languages.